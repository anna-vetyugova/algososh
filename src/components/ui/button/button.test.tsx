import React from 'react';
import {render, screen} from '@testing-library/react'

import { Button } from './button';
import userEvent from '@testing-library/user-event';

describe( 'Button component', () => {
  it('renders without text', () => {
    // arrange
    render(<Button />);
    const buttonElement = screen.getByRole('button');
    
    // act
    // assert
    expect(buttonElement).toMatchSnapshot();
  });
  it('renders with text', () => {
    render(<Button text={'text'} />);
    const buttonElement = screen.getByRole('button');
    expect(buttonElement).toMatchSnapshot();
  });
  it('renders with disable prop', () => {
    render(<Button disabled={true} />);
    const buttonElement = screen.getByRole('button');
    expect(buttonElement).toMatchSnapshot();
    expect(buttonElement).toBeDisabled();
  });
  it('renders with loader prop', () => {
    render(<Button isLoader={true} />);
    const buttonElement = screen.getByRole('button');
    expect(buttonElement).toMatchSnapshot();
  });
  it('is clicked', () => {
    const click = jest.fn();
    render(<Button onClick={click} />);

    const buttonElement = screen.getByRole('button');
    userEvent.click(buttonElement);

    expect(click).toHaveBeenCalled();
  });
});