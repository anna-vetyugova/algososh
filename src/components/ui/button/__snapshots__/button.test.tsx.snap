// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders with disable prop 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button component renders with loader prop 1`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button component renders with text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    text
  </p>
</button>
`;

exports[`Button component renders without text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;
